animation_nodes/nodes/color/vertex_color_input.py

```diff
@@ -0,0 +1,132 @@
+import bpy
+from bpy.props import *
+from ... events import propertyChanged
+from ... base_types import AnimationNode, VectorizedSocket
+from ... data_structures import Color, ColorList, VirtualColorList, LongList
+from . c_utils import getVertexColorsFromLoopColors, getPolygonColorsFromLoopColors
+
+modeItems = [
+    ("ALL", "All", "Get color of every vertex", "NONE", 0),
+    ("INDEX", "Index", "Get color of a specific vertex", "NONE", 1)
```

Comment:
Remove the index mode.

Reply:
Omar, this `Index` mode is to _Get color of a specific vertex_ but not for _Color Layer_. Which one do you want to remove?

Reply:
The node should return a list of colors all the time. So yes, the Index mode is not needed.

Reply:
I want to remove the one that "Get color of a specific vertex".

Reply:
Okay.

Reply:
I have removed the `Index` mode and put the remaing code into a `execute` function,
```
def execute(self, object, identifier):
        if object is None: return
        
        colorLayer = self.getVertexColorLayer(object, identifier)
        defaultColor = Color((0, 0, 0, 1))
        
        colorsList = ColorList(length = len(colorLayer.data))
        colorLayer.data.foreach_get("color", colorsList.asNumpyArray())

        if self.colorMode == "LOOP":
            return colorsList

        elif self.colorMode == "VERTEX":
            sourceMesh = object.an.getMesh(False)
            polygonIndices = sourceMesh.an.getPolygonIndices()
            vertexCount = len(sourceMesh.vertices)
            
            colorsList = VirtualColorList.create(colorsList, defaultColor)
            return getVertexColorsFromLoopColors(vertexCount, polygonIndices, colorsList)

        elif self.colorMode == "POLYGON":
            sourceMesh = object.an.getMesh(False)
            polygonIndices = sourceMesh.an.getPolygonIndices()

            colorsList = VirtualColorList.create(colorsList, defaultColor)
            return getPolygonColorsFromLoopColors(polygonIndices, colorsList)
```